.github/workflows/ci.yml

```
@@ -12,15 +12,13 @@ permissions:
 jobs:
   test:
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python 3.9
-        uses: actions/setup-python@v3
+      - uses: actions/setup-python@v3
         with:
           python-version: "3.9"
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install pytest
+          cache: "pip" # caching pip dependencies
```

nit: the comment isn't conveying any more useful information
```suggestion
          cache: "pip"
```